hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -288,12 +288,9 @@ public static <R> HoodieData<HoodieRecord<R>> mergeForPartitionUpdatesIfNeeded(
     // the tagging partitions and locations
     // NOTE: The incoming records may only differ in record position, however, for the purpose of
     //       merging in case of partition updates, it is safe to ignore the record positions.
-    HoodieData<HoodieRecordGlobalLocation> globalLocations = incomingRecordsAndLocations
+    HoodieData<Pair<String, String>> globalLocations = incomingRecordsAndLocations
         .filter(p -> p.getRight().isPresent())
-        .map(p -> new HoodieRecordGlobalLocation(
-            p.getRight().get().getPartitionPath(),
-            p.getRight().get().getInstantTime(),
-            p.getRight().get().getFileId()))
+        .map(p -> Pair.of(p.getRight().get().getPartitionPath(), p.getRight().get().getFileId()))
```

Comment:
LGTM, but can I ask a question. look like the `instantime` is different with the same fgId in `HoodieRecordGlobalLocation`, is this normal? In `Global Simple`, it will use base file commit time as `InstantTime`, and there is no different timestamp in the same file. So there is a problem here, `HoodieRecordLocation's instantTime` is baseFile lastCommit time which file name contain, or every record commit time? or may be use any of them, there will be no problem.

Reply:
hey @KnightChess : atleast for global simple, global bloom and RLI, instant time should not matter. Hbase might have some intricacies with instant time to deduce rollbacks. I did not want to increase the scope of this patch. We are good wrt global simple, global bloom and RLI w/ cur fix.

Reply:
https://issues.apache.org/jira/browse/HUDI-7800

Reply:
@nsivabalan thanks for answering